zk_prover/src/circuits/tests.rs

```diff
@@ -501,6 +538,55 @@ mod test {
         valid_prover.assert_satisfied();
     }
 
+    #[test]
+    fn test_solvency_on_chain_verifier() {
+        let params = generate_setup_params(10);
+
+        let merkle_sum_tree =
+            MerkleSumTree::<N_ASSETS>::new("src/merkle_sum_tree/csv/entry_16.csv").unwrap();
+
+        let assets_sum = [Fp::from(556863u64), Fp::from(556863u64)];
+
+        let circuit = SolvencyCircuit::<L, N_ASSETS, N_BYTES>::init(merkle_sum_tree, assets_sum);
+
+        let pk = gen_pk(&params, &circuit, None);
+
+        get_verification_cost(&params, &pk, circuit.clone());
+
+        let num_instances = circuit.num_instance();
+        let instances = circuit.instances();
+
+        let proof_calldata = gen_evm_proof_shplonk(&params, &pk, circuit, instances.clone());
+
+        // let yul_code_path = "../contracts/src/SolvencyVerifier.yul";
```

Comment:
It seems there might be some leftovers in this test.

Reply:
Yes, I removed this too in 9b73cd0. in Just pointing it to @alxkzmn too as a way to show how to use the API to write the solidity verifier contract and get the solidity calldata